src/utils/tools/updatingIcons/prepareTheUpdate.ts

```diff
@@ -0,0 +1,28 @@
+import { IconInterface } from 'components/icons'
+
+const prepareUpdatedIcon = (newIconDetails: IconInterface) => {
+  const updateDetails = {
+    $set: {},
+    $addToSet: {}
```

Comment:
`$set` is used for full replacement, this will be used to use all the incoming fields except, `tags`, `categories` since they're an arrays.
`$addToSet` is used for the `tags`, and  `category`, to add the tags or categories that aren't already exist.